### ISSUE TITLE
Add option to pass env variables via CLI

### ISSUE THREAD

**Issue by ismagilov** [svenkubiak/mangooio#291]
If you want to use `masterkey` for values encryption, you need to have an absolute path to a file with `masterkey`. 

As far as I understood, it is impossible to create a file and upload to Heroku's filesystem. You can only set environment variables or a special command line string with parameters.

Do you think it is reasonable to extend the set of variables read from command line with `masterkey`'s variable?

**Comment by svenkubiak** [comment]
Sounds like a good improvement in regards to get in Cloud ready. Just to clearify: this will be a static string, not a file reference, e.g.

-Dmasterkey=thisismymasterkeyt

Is there anything else that heroku might need in terms of variables from the command line?

**Comment by ismagilov** [comment]
> -Dmasterkey=thisismymasterkeyt

Exactly, we need to pass a value, not a file reference.

> Is there anything else that heroku might need in terms of variables from the command line?

Roughly speaking, Heroku takes two artifacts from a developer: an application `jar` and a `procfile`. It launches `jar` using command line from `procfile`.

Before launching it sets default config variables, which you can access using `$VAR` notation in your `procfile`. Besides default config vars, you can set your own. And besides that, some additional Heroku modules can extend the set of config vars with their values. For sample, URL for NoSQL db.

This is how my `procfile` currently looks like: 
```
web: java -Dhttp.host=0.0.0.0 -Dhttp.port=$PORT -jar target/mangooioapp.jar
```
That is why in #271 I asked to add `http.port` option. Heroku defines a port for you externally.

By default, Heroku sets [these variables](https://devcenter.heroku.com/articles/java-support#environment): `PORT`, `JAVA_OPTS`, `DATABASE_URL `.  

Summing up, I would say that those parameter are needed to be set from CLI, which are either known **only** in cloud runtime or their values are **not available** in a cloud (like file paths).

As for variables, like `DATABASE_URL`, I think that they can be passed directly to Java through `-D`, and later read in application through `System.get...`. Of course, all this forces you to configure connections and similar object only in runtime. So if you are going to add persistent level configuration, probably, you should keep it in mind also.

**Comment by svenkubiak** [comment]
Do you get Database credentials and URLs at runtime or before? Before would enable you the oppertunity to pass them in the application.yaml.

**Comment by ismagilov** [comment]
According with docs, situation with database credits is the same:

> ... an application should read the DATABASE_URL environment variable (or JDBC_DATABASE_URL variable described later) and set up the database connections based on that information.

Recommended way is to [read it in runtime using `System.get...`](https://devcenter.heroku.com/articles/connecting-to-relational-databases-on-heroku-with-java#using-the-database_url-in-plain-jdbc):
```
System.getenv("DATABASE_URL")
```

Or, if frameworks where env vars can be injected into config files (Spring Boot, Play) are used, developers can [reference env vars in a config](https://devcenter.heroku.com/articles/connecting-to-relational-databases-on-heroku-with-java#using-the-jdbc_database_url-in-a-spring-boot-app):
```
spring:
  datasource:
    url: ${JDBC_DATABASE_URL}
    username: ${JDBC_DATABASE_USERNAME}
    password: ${JDBC_DATABASE_PASSWORD}
```

**Comment by svenkubiak** [comment]
I think referencing env vars sounds like a good approach on this. We need to split masterkey and masterkeyfile though. 

If we introduce refrencing env vars then I would like to drop the newly introduced http.port, etc. parameters in 5.0.0 as they are ambiguous.
